packages/core/src/browser/quick-input/quick-input-service.ts

```diff
@@ -37,6 +37,7 @@ export interface QuickPickItem {
     description?: string;
     detail?: string;
     keySequence?: KeySequence;
+    resource?: URI;
```

Comment:
I'm not sure the base `quick-input-service` should know about resource URIs, it is specific to the _'open recent workspace...'_ or other file-based quick-inputs.

Reply:
Nice catch!,  a new interface is now introduced for this purpose